### PR TITLE
adding probe error log support 

### PR DESCRIPTION
**Pull request by QingsongYao**
this PR address issue #350 due to we need to keep failed probe history longer for people to investigate issues. debug logging can not address this issue due to people should not need to look at log to troubleshot probe issue. when probing large number of sites, the failed log entry will be override quickly by succeed logs, so we create a separate entry for this.

I have no tested the change yet due to don't know how to setup build and test on local env

### CODE REVIEW

**Comment by brian-brazil** [comment]
I think this would be better combined with the existing list, and reusing the existing flag.

**Comment by QingsongYao** [comment]
It make sense that existing list track succeed requests,  new failed requested list tracked failed requests.  I will reusing the existing flag

**Comment by brian-brazil** [comment]
The existing list tracks all requests.

**Comment by brian-brazil** [comment]
You can have two lists, and merge them.

**Comment by QingsongYao** [comment]
Thank you for the suggestion. I made change to this PR to use two lists and merged them when display to UI.

File: main.go [hunk 1]
```diff
@@ -285,14 +290,18 @@ func main() {
     <h2>Recent Probes</h2>
     <table border='1'><tr><th>Module</th><th>Target</th><th>Result</th><th>Debug</th>`))
 
-		results := rh.List()
+		results := eh.List()
```
Comment (brian-brazil) [inline main.go]: You're not merging the lists, it should be ordered by time.

File: main.go [hunk 2]
```diff
@@ -308,6 +317,9 @@ func main() {
 			return
 		}
 		result := rh.Get(id)
+		if r.URL.Query().Get("error") == "true" {
```
Comment (brian-brazil) [inline main.go]: This should work transparently

File: main.go (hunk 1 above)
Comment (QingsongYao) [reply main.go]: result struct has no datetime field, so it will be hard to merge and sort by time.

Comment (brian-brazil) [reply main.go]: You could use the ids

Comment (QingsongYao) [reply main.go]: The ids in two list might conflicts since both set and increase independently .

Comment (brian-brazil) [reply main.go]: A common counter will handle that

**Comment by SuperQ** [comment]
It would be nice to get this done, being able to log all errors rather than just all results would be pretty useful.

**Comment by brian-brazil** [comment]
Obsoleted by #517

**Pull request closed without merging**